sway-types/src/span.rs

```diff
@@ -71,6 +71,44 @@ impl PartialEq for Span {
 
 impl Eq for Span {}
 
+impl<'de> Deserialize<'de> for Span {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        struct SpanVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for SpanVisitor {
+            type Value = Span;
+        
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                write!(formatter, "Span")
+            }
+
+            fn visit_seq<A: serde::de::SeqAccess<'de>>(self, mut seq: A) -> Result<Self::Value, A::Error> {
+                let mut values: Vec<usize> = vec![];
+
+                while let Some(value) = seq.next_element()? {
+                    values.push(value);
+                }
+
+                if values.len() != 2 {
+                    return Err(serde::de::Error::custom(
+                        format!("Invalid Span sequence: expected 2 elements, found {}", values.len())
+                    ));
+                }
+
+                Ok(Span {
+                    // The `src` field is not serialized, so we have to generate a valid one:
+                    src: Arc::from((0 .. values[0] + values[1]).map(|_| ' ').collect::<String>()),
+                    start: values[0],
+                    end: values[1],
+                    source_id: None,
```

Comment:
This may cause problems depending on how the deserialized AST is run.

@JoshuaBatty @tritao what do you guys think?